modules/apps/callbacks/types/callbacks_test.go

```diff
@@ -0,0 +1,575 @@
+package types_test
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cometbft/cometbft/crypto/secp256k1"
+
+	"github.com/cosmos/ibc-go/v7/modules/apps/callbacks/types"
+	transfer "github.com/cosmos/ibc-go/v7/modules/apps/transfer"
+	transfertypes "github.com/cosmos/ibc-go/v7/modules/apps/transfer/types"
+	channeltypes "github.com/cosmos/ibc-go/v7/modules/core/04-channel/types"
+	ibctesting "github.com/cosmos/ibc-go/v7/testing"
+	ibcmock "github.com/cosmos/ibc-go/v7/testing/mock"
+)
+
+func (s *CallbacksTypesTestSuite) TestGetSourceCallbackDataTransfer() {
```

Comment:
I think the bulk of these test cases could have directly called `getCallbackData` and `computeExecAndCommitGas` (two separate unit tests). If you agree you can open an issue (not important to do it anytime soon)

Reply:
done